src/main/scala/io/qbeast/spark/index/OTreeDataAnalyzer.scala

```diff
@@ -142,41 +167,24 @@ object DoublePassOTreeDataAnalyzer extends OTreeDataAnalyzer with Serializable {
         })
     }
 
-  def getColumnStats(
-      dataFrame: DataFrame,
-      columnTransformers: IISeq[Transformer]): Seq[ColumnStats] = {
-    val dataFrameStats = dataFrame
-      .describe(columnTransformers.map(_.columnName): _*)
-      .collect()
-
-    columnTransformers
-      .map(ct => {
-        val s = dataFrameStats
-          .map(row => {
-            (row.getAs[String](0), row.getAs[String](ct.columnName))
-          })
-          .toMap
-        ColumnStats(s, ct.dataType)
-      })
-  }
-
   override def analyze(
       dataFrame: DataFrame,
       indexStatus: IndexStatus,
       isReplication: Boolean): (DataFrame, TableChanges) = {
 
     // Compute the statistics for the indexedColumns
-    val columnStats = getColumnStats(dataFrame, indexStatus.revision.columnTransformers)
+    val columnTransformers = indexStatus.revision.columnTransformers
+    val dataFrameStats = getDataFrameStats(dataFrame, columnTransformers)
 
-    // Check if the DataFrame is empty
-    if (columnStats.head.count == 0) {
+    if (dataFrameStats.isEmpty) {
```

Comment:
If you are adding the count in the SQL statement, it should never return 0 rows, no? Is this if necessary?

Reply:
True! But I believe if it's empty we should not continue, or it will fail on:
 
`assert(min != null && max != null)`
https://github.com/osopardo1/qbeast-spark/blob/a041269b09f0da4cbf95b92b0771183318663be5/core/src/main/scala/io/qbeast/core/transform/LinearTransformer.scala#L37